src/main/java/com/thealgorithms/dynamicprogramming/UniqueSubsequencesCount.java

```diff
@@ -0,0 +1,53 @@
+package com.thealgorithms.dynamicprogramming;
+
+// Program to find the number of Subsequences can be produced from a string
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+// Author -> https://github.com/Tuhinm2002
+
+public final class UniqueSubsequencesCount {
+
+    private UniqueSubsequencesCount() {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    public static int subseqCount(String str) {
+
+        int[] dp = new int[str.length() + 1];
+        Arrays.fill(dp, -1);
+
+        return recursiveCall(str, 0, dp);
+    }
+
+    public static int recursiveCall(String st, int idx, int[] dp) {

```

Comment:
What do you think to rename method to `countSubsequences`?

Reply:
Actually it is named as per its duty so I kept it like that. But yeah your function perfect for initial function call which is written above

Reply:
I understand that the current name reflects its role, but I believe using more descriptive names like `countSubsequences `helps clarify the method’s purpose at a glance. Many established algorithms like `Quick Sort` or `Merge Sort` don’t emphasize that they’re recursive, but rather focus on what the function achieves. In this case, `countSubsequences `explicitly tells us what the method does, which I think improves readability and maintainability.